src/classes/valuestore.h

```diff
@@ -0,0 +1,35 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+// Copyright (c) 2021 Team Dissolve and contributors
+
+#pragma once
+
+#include "io/import/values.h"
+#include "templates/optionalref.h"
+#include <list>
+#include <vector>
+
+// Value Store
+class ValueStore
+{
+    public:
+    ValueStore() = default;
+    ~ValueStore() = default;
+
+    /*
+     * Data
+     */
+    private:
+    // Value of contained data
+    std::list<std::tuple<std::string, std::vector<double>, ValueImportFileFormat>> data_;
```

Comment:
```suggestion
    std::vector<std::unique_ptr<std::tuple<std::string, std::vector<double>, ValueImportFileFormat>>> data_;
```

I'm guessing that you went with `std::list` because it doesn't have the pointer invalidation issues that `std::vector` has.  I've been finding that `std::vector<std::unique_ptr<>>` tends to get around that issue.  The only part of the interface that would actually have to change is the `data` member function and most of those call sites could be updated fairly trivially.

It may be a pointless micro-optimisation, so I'm not going to fight on it if you have a reason for sticking with `std::list`

Reply:
Ah, now I recall why I used `std::list` (here and for `DataNDStore`s too) - it's a nasty tangle of a problem.  The stored `FileAndFormat` has a `KeywordList` that can't be copied since it stores a bunch of objects derived from `KeywordBase`, which itself is abstract. So using `std::vector` is a no-no since there can be no default-constructed (empty) object.

A `std::vector` of `std::unique_ptr` would work as well as a `std::list`, but given the context of the usage (i.e. in routines that are only used sparingly for testing, and not in production simulations) I would rather leave it for now, and stick another pin in the "Fix KeywordList" voodoo doll.